server/api/models/Progress.js

```diff
@@ -0,0 +1,24 @@
+const mongoose = require("mongoose");
```

Comment:
Looking at this, I feel like maybe we defined the "categories" for measuring progress too strictly. Would it make more sense to have the user define their own categories?